### PR TITLE
FT-04ab:  Bootstrap Slide Modal

### PR DESCRIPTION
**Pull request by Jeysh-git**
In this PR,
   - Created 'users/new' route 
   - Created slide modal component with header,body.
Figma Design Link:
https://www.figma.com/file/i9whcNdWDHjHsXVdpoYyFh/New-Joinee?node-id=1%3A2937

### CODE REVIEW

File: app/routes/index.js [hunk 1]
```diff
@@ -2,6 +2,6 @@ import Route from '@ember/routing/route';
 
 export default Route.extend({
     beforeModel() {
-        this.replaceWith('users');
+        this.replaceWith('users/new');
```
Comment (shenbaga-praburaj) [inline app/routes/index.js]: Why do we need this ?

File: app/routes/users.js [hunk 2]
```diff
@@ -83,6 +83,8 @@ export default Route.extend({
     //     });
     // }
       model(){
-        return this.store.findAll('user') ;
+        return this.store.findAll('user').then(results => results.filter((user)=>{
+          return user.get('isNew')==false
```
Comment (shenbaga-praburaj) [inline app/routes/users.js]: Good. We can simplify this like, `user => !get(user, 'isNew')`
For single line functions, we don't need to explicitly return.

File: app/styles/app.scss [hunk 3]
```diff
@@ -323,10 +333,83 @@ input.light {
...
 
+.buttons {
+   display: flex;
+}
 
+.headertext {
+   padding: 10px;
+}
 
+.button1 {
```
Comment (shenbaga-praburaj) [inline app/styles/app.scss]: .buttons, .button, .side are very generic class names. So, if we need to define module specific css properties, it's better to define with parent class hierarchy.

And, we can avoid names like button1, slide2. Use names like add-btn, save-btn, add-employee-slide etc..

File: app/templates/components/user-avatar.hbs [hunk 4]
```diff
@@ -10,5 +10,5 @@
              {{initial title}} 
         </div>
     {{/if}}
-    {{outlet}}
+    {{!-- {{outlet}} --}}
```
Comment (shenbaga-praburaj) [inline app/templates/components/user-avatar.hbs]: Outlets are not applicable for component. We would directly call other components to render the children.

File: app/routes/index.js (hunk 1 above)
Comment (Jeysh-git) [reply app/routes/index.js]: Its not needed. Updated with previous one.

File: app/styles/app.scss (hunk 3 above)
Comment (Jeysh-git) [reply app/styles/app.scss]: Updated.

File: app/templates/components/slide-modal.hbs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class = "add-employee-slide-container font-style">
+  <div class = "add-employee-outerslide"></div>
+  <div class = "add-employee-slide">
+    <div class = "header slideheader">
+      <div class="slide-headertext">Add Employee</div>
+      <div class="add-employee-slide-btns">
+        <div class="apply-btn">Apply</div>
+        <div class="reset-btn">Reset</div>
```
Comment (shenbaga-praburaj) [inline app/templates/components/slide-modal.hbs]: We can write this component as a common wrapper and send necessary parameters and yield to fit for the add employee functionality. This way, the slide model can be reused for any other module.

And, we would be using same component for edit user as well. So, we can use generic class names.

File: app/styles/app.scss [hunk 6]
```diff
@@ -323,10 +334,81 @@ input.light {
...
+      color: #fff;
+   }
 
+}
+
+.slide-headertext {
+   padding: 10px;
+}
 
+@keyframes slide-modal {
```
Comment (shenbaga-praburaj) [inline app/styles/app.scss]: This file is getting big. We can spit it module separated css files and import them. For example, all slider related css can be written in slider.scss.

**Comment by shenbaga-praburaj** [comment]
Good work @Jeysh-git , Keep it up 👍🏾

**Review by shenbaga-praburaj — APPROVED** [review]

**Pull request merged**